client/src/utils/similarity.test.js

```diff
@@ -0,0 +1,64 @@
+const Similarity = require('./similarity.js')
+
+test('Test 0: properly calculates cosine similarity', () => {
```

Comment:
Can you add descriptive tests? The tests are not currently descriptive. For example this should probably be
```suggestion
test('returns 0 for the same track', () => {
```

You also don't need to index your tests since your test runner will do that for you.